src/main/java/com/mraof/minestuck/network/computer/SkaianetInfoPacket.java

```diff
@@ -2,146 +2,117 @@
+		}
 	}
 	
-	public static SkaianetInfoPacket request(int playerId)
+	public record HasEntered(boolean hasEntered) implements MSPacket.PlayToClient
 	{
-		SkaianetInfoPacket packet = new SkaianetInfoPacket();
-		packet.playerId = playerId;
+		@Override
+		public void encode(FriendlyByteBuf buffer)
+		{
+			buffer.writeBoolean(hasEntered);
+		}
+		
+		public static HasEntered decode(FriendlyByteBuf buffer)
+		{
+			return new HasEntered(buffer.readBoolean());
+		}
 		
-		return packet;
+		@Override
+		public void execute()
+		{
+			SkaiaClient.handlePacket(this);
+		}
 	}
 	
-	@Override
-	public void encode(FriendlyByteBuf buffer)
+	public record Request(int playerId) implements MSPacket.PlayToServer
 	{
-		if(landChains != null) //Land chain data
+		@Override
+		public void encode(FriendlyByteBuf buffer)
+		{
+			buffer.writeInt(this.playerId);
+		}
+		
+		public static Request decode(FriendlyByteBuf buffer)
 		{
-			buffer.writeBoolean(true);
-			for(List<ResourceKey<Level>> list : landChains)
-			{
-				buffer.writeInt(list.size());
-				for(ResourceKey<Level> land : list)
-				{
-					if(land == null)
-						buffer.writeUtf("");
-					else buffer.writeUtf(land.location().toString());
-				}
-			}
-		} else
+			return new Request(buffer.readInt());
+		}
+		
+		@Override
+		public void execute(ServerPlayer player)
 		{
-			buffer.writeBoolean(false);
-			buffer.writeInt(playerId);
-			
-			if(connectionsFrom != null)
-			{
-				
-				buffer.writeBoolean(isClientResuming);
-				buffer.writeBoolean(isServerResuming);
-				
-				buffer.writeInt(openServers.size());
-				for(Map.Entry<Integer, String> entry : openServers.entrySet())
-				{
-					buffer.writeInt(entry.getKey());
-					buffer.writeUtf(entry.getValue(), 16);
-				}
-				
-				for(SburbConnection connection : connectionsFrom)
-					connection.toBuffer(buffer);
-			}
+			ComputerInteractions.requestInfo(player, IdentifierHandler.getById(this.playerId));
 		}
 	}
 	
-	public static SkaianetInfoPacket decode(FriendlyByteBuf buffer)
+	public record LandChains(List<List<ResourceKey<Level>>> landChains) implements MSPacket.PlayToClient
 	{
-		SkaianetInfoPacket packet = new SkaianetInfoPacket();
-		if(buffer.readBoolean())
+		@Override
+		public void encode(FriendlyByteBuf buffer)
 		{
-			packet.landChains = new ArrayList<>();
-			while(buffer.readableBytes() > 0)
-			{
-				int size = buffer.readInt();
-				List<ResourceKey<Level>> list = new ArrayList<>();
-				for(int k = 0; k < size; k++)
-				{
-					String landName = buffer.readUtf(32767);
-					if(landName.isEmpty())
-						list.add(null);
-					else list.add(ResourceKey.create(Registries.DIMENSION, new ResourceLocation(landName)));
-				}
-				packet.landChains.add(list);
-			}
-		} else
+			buffer.writeCollection(landChains, (buffer1, list) ->
+					buffer1.writeCollection(list, (buffer2, land) ->
+							buffer2.writeOptional(Optional.ofNullable(land),
+									FriendlyByteBuf::writeResourceKey)));
```

Comment:
While very efficient in terms of the number of lines it takes up, it takes a while to unpack. Extraction to a method that names what is going on or just adding a comment may be helpful

Reply:
True. Lambdas within lambdas like this is not the best. As you say, method extraction might be suitable here.